### PR TITLE
ref: only support Python 3.6+

### PR DESCRIPTION
**Pull request by joshuarli**
Following up to https://github.com/getsentry/sentry/pull/23342/#pullrequestreview-576736075, this removes the following custom checks that no longer apply for py2/3 compatability: b102, b301, b302, b303, b304, b305, b306, b307, b308, b309, b310, b311, b313, b315, b316.

And also updates CI and build stuff. And removes Travis. We'll figure out how to publish with GHA later. Could use craft?

I also upgraded flake8 to 3.8.x, and installed this branch into my sentry and getsentry venv and `pre-commit run flake8 --all-files` (pre-commit uses the venv's flake8) reveals some new useful things which I will fix when we cut new sentry-flake8 release.

TODO: submit PR to sentry to ignore these error codes ASAP. Then follow up with upgrading sentry-flake8 after this lands. Before we cut a new release (I'm thinking 1.0) we should also lint against __future__.

sentry

```
src/sentry/db/deletion.py:25:17: F523 '...'.format(...) has unused arguments at position(s): 2
src/sentry/integrations/cloudflare/webhook.py:227:26: F523 '...'.format(...) has unused arguments at position(s): 0
src/sentry/utils/distutils/commands/build_assets.py:62:22: F523 '...'.format(...) has unused arguments at position(s): 0
src/sentry/models/user.py:339:1: E302 expected 2 blank lines, found 1
tests/sentry/integrations/github/test_webhooks.py:33:15: F523 '...'.format(...) has unused arguments at position(s): 0
tests/sentry/api/endpoints/test_sentry_app_installation_external_requests.py:82:17: F523 '...'.format(...) has unused arguments at position(s): 1
tests/sentry/api/endpoints/test_sentry_app_installation_external_requests.py:88:26: F523 '...'.format(...) has unused arguments at position(s): 1
tests/sentry/integrations/github_enterprise/test_webhooks.py:45:15: F523 '...'.format(...) has unused arguments at position(s): 0
src/sentry/identity/base.py:19:41: F523 '...'.format(...) has unused arguments at position(s): 0
tests/sentry/api/endpoints/test_sentry_app_interaction.py:103:14: F507 '...' % ... has 1 placeholder(s) but 2 substitution(s)
tests/sentry/api/endpoints/test_sentry_app_interaction.py:117:14: F507 '...' % ... has 1 placeholder(s) but 2 substitution(s)
src/sentry/utils/batching_kafka_consumer.py:38:21: F523 '...'.format(...) has unused arguments at position(s): 0, 1
```

getsentry

```
tests/acceptance/test_orgredirect.py:13:21: F523 '...'.format(...) has unused arguments at position(s): 0
tests/getsentry/api/test_customer_index.py:138:36: F523 '...'.format(...) has unused arguments at position(s): 1
tests/getsentry/api/test_customer_index.py:143:36: F523 '...'.format(...) has unused arguments at position(s): 1
tests/getsentry/api/test_customer_index.py:148:36: F523 '...'.format(...) has unused arguments at position(s): 1
tests/getsentry/api/test_customer_index.py:156:36: F523 '...'.format(...) has unused arguments at position(s): 1
tests/getsentry/api/test_customer_index.py:161:36: F523 '...'.format(...) has unused arguments at position(s): 1
tests/getsentry/api/test_customer_index.py:170:36: F523 '...'.format(...) has unused arguments at position(s): 1
tests/getsentry/api/test_customer_index.py:175:36: F523 '...'.format(...) has unused arguments at position(s): 1
tests/getsentry/api/test_customer_index.py:185:36: F523 '...'.format(...) has unused arguments at position(s): 1
tests/getsentry/api/test_customer_index.py:191:36: F523 '...'.format(...) has unused arguments at position(s): 1
tests/getsentry/api/test_customer_index.py:196:36: F523 '...'.format(...) has unused arguments at position(s): 1
tests/getsentry/api/test_customer_index.py:207:36: F523 '...'.format(...) has unused arguments at position(s): 1
tests/getsentry/api/test_customer_index.py:211:36: F523 '...'.format(...) has unused arguments at position(s): 1
tests/getsentry/api/test_customer_index.py:217:36: F523 '...'.format(...) has unused arguments at position(s): 1
tests/getsentry/api/test_customer_index.py:222:36: F523 '...'.format(...) has unused arguments at position(s): 1
tests/getsentry/api/test_customer_index.py:234:36: F523 '...'.format(...) has unused arguments at position(s): 1
tests/getsentry/api/test_customer_index.py:238:36: F523 '...'.format(...) has unused arguments at position(s): 1
tests/getsentry/api/test_customer_index.py:246:36: F523 '...'.format(...) has unused arguments at position(s): 1
tests/getsentry/api/test_customer_index.py:251:36: F523 '...'.format(...) has unused arguments at position(s): 1
getsentry/api/sponsored_account_request.py:34:29: E741 ambiguous variable name 'l'
```

### CODE REVIEW

File: src/sentry_check.py [hunk 1]
```diff
@@ -17,15 +17,13 @@ def __init__(self, filename, lines):
         self.filename = filename
         self.lines = lines
 
-        self.has_absolute_import = False
-        self.satisfies_B315_imports = False
+        # self.has_absolute_import = False
```
Comment (wedamija) [inline src/sentry_check.py:20]: Delete?

File: src/sentry_check.py [hunk 2]
```diff
@@ -40,33 +38,18 @@ def visit_ExceptHandler(self, node):
...
-        if node.module == "itertools":
-            for nameproxy in node.names:
-                if nameproxy.name in B316.names:
-                    self.itertools_izip = True
-                    break
+        # if node.module == "__future__":
+        #     for nameproxy in node.names:
+        #         if nameproxy.name == "absolute_import":
+        #             self.has_absolute_import = True
+        #             break
```
Comment (wedamija) [inline src/sentry_check.py:52]: Delete?

**Review by wedamija — COMMENTED** [review]
> And also updates CI and build stuff. And removes Travis. We'll figure out how to publish with GHA later. Could use craft?

What publishing does this break?

File: src/sentry_check.py (hunk 1 above)
Comment (joshuarli) [reply src/sentry_check.py:20]: Nah, I plan on following up using some of this to implement __future__ disallow.

File: src/sentry_check.py (hunk 2 above)
Comment (joshuarli) [reply src/sentry_check.py:52]: Nah, I plan on following up using some of this to implement __future__ disallow.

**Comment by joshuarli** [comment]
> > And also updates CI and build stuff. And removes Travis. We'll figure out how to publish with GHA later. Could use craft?
> 
> What publishing does this break?

Generally we want to remove Travis right? After this, pushing a tag will no longer cause Travis to automatically release. I figure we should be setting up Craft for things like this.

**Comment by wedamija** [comment]
> > > And also updates CI and build stuff. And removes Travis. We'll figure out how to publish with GHA later. Could use craft?
> > 
> > 
> > What publishing does this break?
> 
> Generally we want to remove Travis right? After this, pushing a tag will no longer cause Travis to automatically release. I figure we should be setting up Craft for things like this.

I guess I'm just checking that this doesn't break some workflow for the OSS team

**Comment by evanpurkhiser** [comment]
IMO I would leave travis in until we're ready to migrate it.

**Review by wedamija — APPROVED** [review]

**Pull request merged**